internal/storage/block_storage.go

```diff
@@ -490,14 +490,21 @@ func (b *BlockStorage) UpdateBalance(
 	}
 
 	var existingValue string
-	if exists {
+	switch {
+	// this could happen if balances are bootstrapped and should not be overridden
```

Comment:
Nit: we should capitalize the first letter in all comments on their own line (should've done this in the suggestions...didn't know you were going to use directly)

Reply:
Fixed. :+1: